std/typecons.d

```diff
@@ -1801,7 +1810,15 @@ Forces $(D this) to the null state.
  */
     void nullify()()
     {
-        .destroy(_value);
+        static if (defaultConstructed)
```

Comment:
There is actually no such thing as `defaultConstructed` in D (as of now). Both pieces of code are valid here (since spec says a class has to always support destroying it's T.init). But I'd still favor the option of only destroying non-null things, as that is what makes most sense.